src/Product/ProductRepository.php

```diff
@@ -271,7 +281,39 @@ public function find_mc_not_synced_product_ids( int $limit = -1, int $offset = 0
 			],
 		];
 
-		return $this->find_ids( $args, $limit, $offset );
+		$not_synced_ids = $this->find_ids( $args, $limit, $offset );
+
+		// Get only the product that is syncable from 'not synced' products,
+		// including variation products of a variable product.
+		$product_ids = array_filter(
+			$not_synced_ids,
+			function ( $product_id ) {
+				$product = $this->product_helper->get_wc_product( $product_id );
+
+				// If it's a variable product, check its children variation products.
+				if ( $product instanceof WC_Product_Variable ) {
+					if ( empty( $product->get_children() ) ) {
+						// Return false if no variation products. I.e. it's not syncable.
+						return false;
+					} else {
+						// Get a list of variation products, return true if any of a varation product
+						// is syncable, otherwise return false.
+						foreach ( $product->get_children() as $variation_id ) {
+							$variation = $this->product_helper->get_wc_product( $variation_id );
+							if ( $this->product_helper->is_sync_ready( $variation ) ) {
+								return true;
+							}
+						}
+						return false;
+					}
+				}
+
+				// Check if a simple product is syncable.
+				return $this->product_helper->is_sync_ready( $product );
```

Comment:
The `is_sync_ready` check here and for variable products will exclude some products but we have built in compatibility for bundles which is attached to `woocommerce_gla_get_sync_ready_products_pre_filter`. That means they'll be excluded from the syncable products, but not excluded here.

Reply:
Thanks for pointing this out, this is something I've missed.